cms/djangoapps/contentstore/views/item.py

```diff
@@ -843,6 +844,40 @@ def safe_get_username(user_id):
     return xblock_info
 
 
+def add_container_page_publishing_info(xblock, xblock_info):  # pylint: disable=invalid-name
+    """
+    Adds information about the xblock's publish state to the supplied
+    xblock_info for the container page.
+    """
+    def safe_get_username(user_id):
+        """
+        Guard against bad user_ids, like the infamous "**replace_user**".
+        Note that this will ignore our special known IDs (ModuleStoreEnum.UserID).
+        We should consider adding special handling for those values.
+
+        :param user_id: the user id to get the username of
+        :return: username, or None if the user does not exist or user_id is None
+        """
+        if user_id:
+            try:
+                return User.objects.get(id=user_id).username
+            except:  # pylint: disable=bare-except
```

Comment:
Why are we doing a bare `except` here? At the very least, we should be doing `except Exception:`, but it would still be much better to actually catch the exceptions that we expect might happen. This [diaper pattern](http://mike.pirnat.com/2009/05/09/the-diaper-pattern-stinks/) is going to make this much, much harder to debug in the future.